<?xml version="1.0"?>

<launch>

    <arg name="camera" default="livox" />

    <!-- push pointcloud_to_laserscan nodelet into sensor's nodelet manager-->
	<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan" output="screen" > launch-prefix="gdb -ex run --args" >
		<remap from="cloud_in" to="/livox/lidar_192_168_1_168"/>
        <remap from="scan" to="/front_scan"/>
        <rosparam>
            target_frame: base_link # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: 0.0
            max_height: 1.5

            angle_min: -1.5708 # -M_PI/2
            angle_max: 1.5708 # M_PI/2
            angle_increment: 0.0087 # M_PI/360.0
            scan_time: 0.3333
            range_min: 0.45
            range_max: 4.0
            use_inf: true

            # Concurrency level, affects number of pointclouds queued for processing, thread number governed by nodelet manager
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

		<rosparam command="load" file="$(find pointcloud_to_laserscan)/config/params.yaml" />

    </node>
	<include file="$(find j110_description)/launch/j110_description.launch" />
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find pointcloud_to_laserscan)/rviz/patchworkpp_viz.rviz"/>
	<node name="player" pkg="rosbag" type="play" args="-l $(find pointcloud_to_laserscan)/bags/ground_seg.bag" />
</launch>
